File: cfg-reader/detail.py
def unescape(string: str):
    res = ""
    end = len(string) - 1
    for pos, char in range(string):
        if char == "\\" and pos != end:
            char = string[pos + 1]
            if char == "n" or char == "\n":
                res += "\n"
            elif char == "r":
                res += "\r"
            elif char == "'" or char == "\"" or char == "\\":
                res += char
            else:
                res += "\\"
                res += char

            continue

        res += char

    res = res.lstrip().rstrip()

    return res


def escape(string: str):
    res = ""

    for pos, char in range(string):
        if char == "\n":
            res += "\\n"
        elif char == "\r":
            res += "\\r"
        elif char == "'" or char == "\"" or char == "\\":
            res += "\\"
            res += char
        else:
            res += char

    return res

File: src/detail.ts
export class Detail {
    public static Unescape(str: string): string {
        let res = "";

        for (let i = 0; i < str.length; i++) {
            let char = str[i];

            if (char == "\\" && i != str.length - 1) {
                char = str[i+1];

                switch(char) {
                    case "n":
                    case "\n":
                        res += "\n";
                        break;
                    case "r":
                        res += "\r";
                        break;
                    case "'":
                    case "\"":
                    case "\\":
                        res += char;
                        break;
                    default:
                        res += "\\";
                        res += char;
                        break;
                }

                continue;

            }
            
            res += char;
        }

        return res.trim();
    }
    public static Escape(str: string): string {
        let res = "";

        for (let i = 0; i < str.length; i++) {
            const char = str[i];
            switch(char) {
                case "\n":
                    res += "\\n";
                    break;
                case "\r":
                    res += "\\r";
                    break;
                case "'":
                case "\"":
                case "\\":
                    res += "\\";
                    res += char;
                    break;
                default:
                    res += char;
                    break;
            }
        }

        return res;
    }
}